gencred/README.md

```diff
@@ -1,241 +1,60 @@
- 
-```console
-$ gencred --name mycluster --output ./config.yaml --serviceaccount
-```
-
-The kubeconfig contents will be `output` to  `./config.yaml`:
+The resulting `new-kubeconfig.yaml` file will look something like:
 
 ```yaml
 apiVersion: v1
 clusters:
 - cluster:
-    certificate-authority-data: fake-ca-data
-    server: https://1.2.3.4
-  name: mycluster
+    certificate-authority-data: same-as-target-context-cluster-info
+    server: https://example.com
+  name: new-context
 contexts:
 - context:
-    cluster: mycluster
-    user: mycluster
-  name: mycluster
-current-context: mycluster
+    cluster: new-context
+    user: new-context
+  name: new-context
 kind: Config
 preferences: {}
 users:
-- name: mycluster
+- name: new-context
   user:
     token: fake-token
 ```
 
-Create a kubeconfig entry with **default** context name `build` using `certificate` authorization and output to the **default** `stdout`.
-
-```console
-$ gencred --certificate
-```
-
-The kubeconfig contents will be `output` to `stdout`:
-
-```yaml
-apiVersion: v1
-clusters:
-- cluster:
-    certificate-authority-data: fake-ca-data
-    server: https://1.2.3.4
-  name: build
-contexts:
-- context:
-    cluster: build
-    user: build
-  name: build
-current-context: build
-kind: Config
-preferences: {}
-users:
-- name: build
-  user:
-    client-certificate-data: fake-cert-data
-    client-key-data: fake-key-data
-```
-
-Specify an alternative kube `context` to generate credentials for.
-
-```console
-$ gencred --context gke_project_us-west1-a_prow
-```
-
-Specify the `--overwrite` flag to *replace* the `output` file if it exists.
-
-```console
-$ gencred --output ./existing.yaml --overwrite
-```
-
-Omit the `--overwrite` flag to *merge* the `output` file if it exists.
-> Entries from the *existing* file take precedence on conflicts.
-
-```console
-$ gencred --name oldcluster --output ./existing.yaml
-$ gencred --name newcluster --output ./existing.yaml
-```
-
-The kubeconfig contents will be `output` to  `./existing.yaml`:
-
-```yaml
-apiVersion: v1
-clusters:
-- cluster:
-    certificate-authority-data: fake-ca-data
-    server: https://1.2.3.4
-  name: oldcluster
-- cluster:
-    certificate-authority-data: fake-ca-data
-    server: https://1.2.3.4
-  name: newcluster
-contexts:
-- context:
-    cluster: oldcluster
-    user: oldcluster
-  name: oldcluster
-- context:
-    cluster: newcluster
-    user: newcluster
-  name: newcluster
-users:
-- name: oldcluster
-  user:
-    client-certificate-data: fake-cert-data
-    client-key-data: fake-key-data
-- name: newcluster
-  user:
-    client-certificate-data: fake-cert-data
-    client-key-data: fake-key-data
-```
-
-### Library
-
-#### Generate a service account token for a cluster. 
-✅ **PREFERRED** method.
-
-```go
-// Import serviceaccount
-import "k8s.io/test-infra/gencred/pkg/serviceaccount"
+The most relevant parts here are `token:` and `certificate-authority-data:`.
```

Comment:
The `context` section seems equally relevant. It is unclear to me why this is included at all.